consensus/state_processing/src/common/initiate_validator_exit.rs

```diff
@@ -8,12 +8,12 @@ pub fn initiate_validator_exit<E: EthSpec>(
     index: usize,
     spec: &ChainSpec,
 ) -> Result<(), Error> {
-    // We do things in a slightly different order to the spec here. Instead of immediately checking
-    // whether the validator has already exited, we instead prepare the exit cache and compute the
-    // cheap-to-calculate values from that. *Then* we look up the validator a single time in the
-    // validator tree (expensive), make the check and mutate as appropriate. Compared to the spec
-    // ordering, this saves us from looking up the validator in the validator registry multiple
-    // times.
+    let validator = state.get_validator_cow(index)?;
+
+    // Return if the validator already initiated exit
+    if validator.exit_epoch != spec.far_future_epoch {
+        return Ok(());
+    }
```

Comment:
I don't know. @realbigsean deleted it in this commit: f4907ef971973b67fb692cb42a5119b5402a4371

Reply:
I needed to do this to fix some EF tests.  In Electra we are mutating the state when calculating `exit_queue_epoch` because we call `compute_exit_epoch_and_update_churn`. So we need to actually do the check for whether the validator exited before calculating `exit_queue_epoch`. 

I figured it wasn't worth trying to maintain the pre-fork optimization at the expense of clarity. Is it worth adding back or trying to find a similar post-fork optimization?